test/lib/dockerfile.spec.ts

```diff
@@ -0,0 +1,269 @@
+          baseImage: "node:dubnium",
+          dockerfilePackages: {},
+          dockerfileLayers: {},
+          error: undefined,
+        },
+      },
+    ],
+    [
+      "a Dockerfile with multiple ARGs and multiple occurrences",
+      {
+        fixture: "with-args-occurences",
+        expected: {
+          baseImage: "test:test-1",
+          dockerfilePackages: {},
+          dockerfileLayers: {},
+          error: undefined,
+        },
+      },
+    ],
+    [
+      "a Dockerfile with ARG for package",
+      {
+        fixture: "with-args-package",
+        expected: {
+          baseImage: "ruby:2.5-alpine",
+          dockerfilePackages: {
+            bash: {
+              instruction:
+                "RUN apk update && apk upgrade && apk add --update --no-cache nodejs bash",
+            },
+            nodejs: {
+              instruction:
+                "RUN apk update && apk upgrade && apk add --update --no-cache nodejs bash",
+            },
+          },
+          dockerfileLayers: {
+            UlVOIGFwayB1cGRhdGUgJiYgYXBrIHVwZ3JhZGUgJiYgYXBrIGFkZCAtLXVwZGF0ZSAtLW5vLWNhY2hlIG5vZGVqcyBiYXNo: {
+              instruction:
+                "RUN apk update && apk upgrade && apk add --update --no-cache nodejs bash",
+            },
+          },
+          error: undefined,
+        },
+      },
+    ],
+    [
+      "a Dockerfile with an installation instruction",
+      {
+        fixture: "with-installation-instruction",
+        expected: {
+          baseImage: "ubuntu:bionic",
+          dockerfileLayers: {
+            UlVOIGFwdC1nZXQgaW5zdGFsbCBjdXJs: {
+              instruction: "RUN apt-get install curl",
+            },
+          },
+          dockerfilePackages: {
+            curl: {
+              instruction: "RUN apt-get install curl",
+            },
+          },
+          error: undefined,
+        },
+      },
+    ],
+    [
+      "multi stage Dockerfile with lowercase instructions",
+      {
+        fixture: "multi-stage-lowercase",
+        expected: {
+          baseImage: "alpine:latest",
+          dockerfilePackages: {
+            "ca-certificates": {
+              instruction: "RUN apk --no-cache add ca-certificates",
+            },
+          },
+          dockerfileLayers: {
+            "UlVOIGFwayAtLW5vLWNhY2hlIGFkZCBjYS1jZXJ0aWZpY2F0ZXM=": {
+              instruction: "RUN apk --no-cache add ca-certificates",
+            },
+          },
+          error: undefined,
+        },
+      },
+    ],
+  ];
+  // tslint:disable-next-line: variable-name
+  test.each<TestCaseTuple>(cases)("%s", async (_description, item) => {
+    const { fixture, expected } = item;
+    const pathToDockerFile = getDockerfileFixture(fixture);
+    const actual = await dockerFile.readDockerfileAndAnalyse(pathToDockerFile);
+    expect(actual.baseImage).toEqual(expected.baseImage);
+    expect(actual.error).toEqual(expected.error);
+    Object.keys(expected.dockerfilePackages).forEach((pkg) => {
```

Comment:
🙀  looping expects inside `.each` -- how does the failure message look when something fails, and is it easy enough to reason with and understand the issue?